### ISSUE TITLE
Allow showing a static image from the frontend folder using Image

### ISSUE THREAD

**Issue by OlliTietavainenVaadin** [vaadin/flow#4849]
Currently, there's no good way of setting the `src` attribute of an Image component to point to a file in the the `/frontend/` directory because of the different ES version directories in production bundles. `vaadinSession.getService().resolveResource()` seems like it could work for this purpose, but apparently it's meant for internal use and it won't handle non-empty context paths correctly in this case.

**Comment by pleku** [comment]
So the issue is really to _make it easy to handle a resource url when the app is not in root context and it is transpiled_ ?

The code that does the same for all other frontend resources should be valid also for this case, but probably is not public.

**Comment by denis-anisimov** [comment]
> So the issue is really to _make it easy to handle a resource url when the app is not in root context and it is transpiled_ ?

I think so.

We probably need a method in `Element` or a helper method which may be used to set `URL` like value. For such cases (which still has to be identified by the component developer) this helper method should be used to set attribute/property value.
Only in those cases the value will be automatically converted if it starts with  `frontend://` or `context://`.

I'm pretty sure that we should not do a conversation for any attribute/property value.

**Comment by OlliTietavainenVaadin** [comment]
> So the issue is really to _make it easy to handle a resource url when the app is not in root context and it is transpiled_ ?

Yes. In other words, make it so that resource urls can be used in a typical production setting.
